core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateV1ToInstanceTemplateV4Converter.java

```diff
@@ -25,7 +28,10 @@ public InstanceTemplateV4Request convert(InstanceTemplateV1Request source, Detai
         response.setAttachedVolumes(getIfNotNull(source.getAttachedVolumes(), volumeConverter::convertTo));
         response.setEphemeralVolume(getIfNotNull(source.getEphemeralVolume(), volumeConverter::convert));
         response.setAws(getIfNotNull(source.getAws(), instanceTemplateParameterConverter::convert));
-        response.setAzure(getIfNotNull(source.getAzure(), environment, instanceTemplateParameterConverter::convert));
+        AzureInstanceTemplateV1Parameters azureParametersEffective = Objects.requireNonNullElse(source.getAzure(), new AzureInstanceTemplateV1Parameters());
+        // InstanceTemplateParameterConverter.convert(AzureInstanceTemplateV1Parameters, DetailedEnvironmentResponse) has to be always invoked for sake of
+        // Azure disk encryption!
```

Comment:
do we really need this comment? commit message and unit test should be enough to ensure we don't delete the code which is necessary. maybe some higher level test to make sure

Reply:
Makes sense. Will remove the comment in a moment.

Reply:
Done.